python/paddle/fluid/tests/unittests/op_test.py

```diff
@@ -351,13 +357,132 @@ def _calc_output(self, place, parallel=False, no_check_set=None, loss=None):
+
+    def check_inplace_grad_output_with_place(self, place, no_check_set=None):
+        # create froward program to get forward vars
+        program = Program()
+        block = program.global_block()
+        op = self._append_ops(block)
+        inputs = self._get_inputs(block)
+        outputs = self._get_outputs(block)
+        feed_map = self.feed_var(inputs, place)
+
+        # get grad_op 
+        if not core._has_grad_op_maker(op.desc.type()):
+            return
+        grad_op_desc_list, op_grad_to_var = core.get_grad_op_desc(op.desc,
+                                                                  set(), [])
+        # has grad_op_maker but no grad_op ?
+        if not grad_op_desc_list:
+            return
+        grad_op_desc = grad_op_desc_list[0]
+
+        # grad_op can not inplace
+        if not core._has_infer_inplace(grad_op_desc.type()):
+            return
+
+        # get forward outs
+        forward_outs, fetch_list = self._calc_output(
+            place, no_check_set=no_check_set)
+
+        # create grad program
+        grad_program = Program()
+        grad_block = grad_program.global_block()
+        new_op_desc = grad_block.desc.append_op()
+        new_op_desc.copy_from(grad_op_desc)
+        grad_program._sync_with_cpp()
+
+        # create grad vars based on forward vars (shape and dtype)
+        for arg in grad_op_desc.input_arg_names(
+        ) + grad_op_desc.output_arg_names():
+            forward_var_name = op_grad_to_var.get(arg, None)
+            if forward_var_name is None:
+                forward_var_name = arg
+            forward_var = block.vars.get(forward_var_name)
+            assert forward_var is not None, "{} cannot be found".format(
+                forward_var_name)
+            grad_var = grad_block.create_var(
+                name=arg,
+                dtype=forward_var.dtype,
+                shape=forward_var.shape,
+                type=forward_var.type,
+                persistable=False)
+        grad_program._sync_with_cpp()
+        grad_fetch_list = grad_op_desc.output_arg_names()
+
+        def _calc_grad_output(enable_inplace=None):
+            # generate feed_map for grad_program
+            # since we don`t really check gradient accuracy, but the consistency when using and not using inplace
+            # we use forward outs (also inputs sometimes) as grad (fake) feeds
+            grad_feed_map = {}
+            for arg in grad_op_desc.input_arg_names():
+                if arg in feed_map.keys():
+                    grad_feed_map[arg] = feed_map[arg]._copy()
+                else:
+                    forward_var_name = op_grad_to_var.get(arg, None)
+                    if forward_var_name is None:
+                        forward_var_name = arg
+                    for i, out in enumerate(fetch_list):
+                        if out.name == forward_var_name:
+                            grad_feed_map[arg] = forward_outs[i]._copy()
```

Comment:
Please `_copy` with place. Re-design the pybind method `_copy` with adding `place` parameter.

Reply:
Done, thanks.